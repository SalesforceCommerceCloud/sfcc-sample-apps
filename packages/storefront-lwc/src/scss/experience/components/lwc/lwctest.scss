.lwctest {
    color: chocolate;
    background-color: azure;
}
