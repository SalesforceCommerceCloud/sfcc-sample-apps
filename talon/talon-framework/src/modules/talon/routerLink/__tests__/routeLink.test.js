import { createElement } from 'lwc';
import RouterLink from 'talon/routerLink';

jest.mock('talon/routingService', () => ({
    getRouteUrl(route, routeParams, queryParams) {
        if (routeParams) {
            Object.entries(routeParams).forEach(([k, v]) => {
                route = route.replace(new RegExp(':' + k, 'g'), v);
            });
        }
        if (queryParams) {
            route += '?';
            Object.entries(queryParams).forEach(([k, v], idx) => {
                if (idx > 0) {
                    route += '&';
                }
                route += `${k}=${v}`;
            });
        }
        return `/${route}`;
    }
}));

describe('talon/routerLink', () => {
    beforeEach(() => {
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    it('renders correctly', async () => {
        const element = createElement('talon-router-link', { is: RouterLink });
        element.label = 'Home';
        element.route = 'home';

        document.body.appendChild(element);

        expect(element).toMatchSnapshot();
    });

    it('renders route params correctly', async () => {
        const element = createElement('talon-router-link', { is: RouterLink });
        element.label = 'Home';
        element.route = 'user/:userId';
        element.routeParams = {
            userId: '123',
            optional: '456'
        };

        document.body.appendChild(element);

        expect(element).toMatchSnapshot();
    });

    it('renders query params correctly', async () => {
        const element = createElement('talon-router-link', { is: RouterLink });
        element.label = 'Home';
        element.route = 'home';
        element.queryParams = {
            p1: '123',
            p2: '456'
        };

        document.body.appendChild(element);

        expect(element).toMatchSnapshot();
    });
});
