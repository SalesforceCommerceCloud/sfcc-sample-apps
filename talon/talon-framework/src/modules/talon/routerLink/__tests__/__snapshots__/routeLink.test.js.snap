// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`talon/routerLink renders correctly 1`] = `
<talon-router-link>
  #shadow-root(open)
    <a
      href="/home"
    >
      Home
    </a>
</talon-router-link>
`;

exports[`talon/routerLink renders query params correctly 1`] = `
<talon-router-link>
  #shadow-root(open)
    <a
      href="/home?p1=123&p2=456"
    >
      Home
    </a>
</talon-router-link>
`;

exports[`talon/routerLink renders route params correctly 1`] = `
<talon-router-link>
  #shadow-root(open)
    <a
      href="/user/123"
    >
      Home
    </a>
</talon-router-link>
`;
